nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/groovy/org/apache/nifi/security/util/crypto/KeyedEncryptorGroovyTest.groovy

```diff
@@ -1,254 +0,0 @@
-        InputStream plainStream = new ByteArrayInputStream(PLAINTEXT.getBytes("UTF-8"))
-
-        OutputStream cipherStream = new ByteArrayOutputStream()
-        OutputStream recoveredStream = new ByteArrayOutputStream()
-
-        EncryptionMethod encryptionMethod = EncryptionMethod.AES_CBC
-        logger.info("Using ${encryptionMethod.name()}")
-
-        // Act
-        KeyedEncryptor encryptor = new KeyedEncryptor(encryptionMethod, KEY)
-
-        StreamCallback encryptionCallback = encryptor.getEncryptionCallback()
-        StreamCallback decryptionCallback = encryptor.getDecryptionCallback()
-
-        encryptionCallback.process(plainStream, cipherStream)
-
-        final byte[] cipherBytes = ((ByteArrayOutputStream) cipherStream).toByteArray()
-        logger.info("Encrypted: {}", Hex.encodeHexString(cipherBytes))
-        InputStream cipherInputStream = new ByteArrayInputStream(cipherBytes)
-        decryptionCallback.process(cipherInputStream, recoveredStream)
-
-        // Assert
-        byte[] recoveredBytes = ((ByteArrayOutputStream) recoveredStream).toByteArray()
-        String recovered = new String(recoveredBytes, "UTF-8")
-        logger.info("Recovered: {}\n\n", recovered)
-        assert PLAINTEXT.equals(recovered)
-    }
-
-    @Test
-    void testShouldDecryptOpenSSLUnsaltedCipherTextWithKnownIV() throws Exception {
-        // Arrange
-        final String PLAINTEXT = new File("${TEST_RESOURCES_PREFIX}/plain.txt").text
-        logger.info("Plaintext: {}", PLAINTEXT)
-        byte[] cipherBytes = new File("${TEST_RESOURCES_PREFIX}/unsalted_128_raw.enc").bytes
-
-        final String keyHex = "711E85689CE7AFF6F410AEA43ABC5446"
-        final String ivHex = "842F685B84879B2E00F977C22B9E9A7D"
-
-        InputStream cipherStream = new ByteArrayInputStream(cipherBytes)
-        OutputStream recoveredStream = new ByteArrayOutputStream()
-
-        final EncryptionMethod encryptionMethod = EncryptionMethod.AES_CBC
-        KeyedEncryptor encryptor = new KeyedEncryptor(encryptionMethod, new SecretKeySpec(Hex.decodeHex(keyHex as char[]), "AES"), Hex.decodeHex(ivHex as char[]))
-
-        StreamCallback decryptionCallback = encryptor.getDecryptionCallback()
-        logger.info("Cipher bytes: ${Hex.encodeHexString(cipherBytes)}")
-
-        // Act
-        decryptionCallback.process(cipherStream, recoveredStream)
-
-        // Assert
-        byte[] recoveredBytes = ((ByteArrayOutputStream) recoveredStream).toByteArray()
-        String recovered = new String(recoveredBytes, "UTF-8")
-        logger.info("Recovered: {}", recovered)
-        assert PLAINTEXT.equals(recovered)
-    }
-
-    /**
-     * This test demonstrates that if incoming cipher text was generated by a cipher using PBE with
-     * KDF, the salt can be skipped and the cipher bytes can still be decrypted using keyed encryption.
-     * @throws Exception
-     */
-    @Test
-    void testShouldSkipSaltOnDecrypt() throws Exception {
```

Comment:
Do we need an equivalent Java test for this and testShouldParseCostParams()?

Reply:
Thanks for the feedback @thenatog!

The new `testEncryptDecrypt` method handles encryption and decryption without a salt specified, so maintaining this particular test seems unnecessary. The `PasswordBasedEncryptorTest` includes several methods for different algorithms, some of which use a salt, so this concept seems covered. This particular method also seems unnecessary since the lower-level Cipher Provider tests already exercise a number of different scenarios.

The `testShouldParseCostParams()` was directly related to this test method, and only exercised another method in this test class, so it is also unnecessary after refactoring.